cdm/core/src/main/java/ucar/nc2/filter/Classifier.java

```diff
@@ -46,25 +74,64 @@ public int[] classifyDoubleArray(Array arr) {
     return classifiedArray;
   }
 
-
-
   /** for a single double */
   public int classifyArray(double val) {
     if (val >= 0) {
```

Comment:
Is this `classifyArray` still being used? maybe we can delete this function as it was the placeholder version and add the categories to the ncml in `testClassifier.ncml` so your original tests keep working. It's possible that this is causing some of the test issues as the original tests may be using this code path and you only added the tolerance to the comparison below in `classifyArrayAttribute`